src/State.cpp

```diff
@@ -483,13 +483,14 @@ namespace polyfem
 
 		logger().info("Extracting boundary mesh...");
 		extract_boundary_mesh();
+		extract_boundary_mesh_pressure();
 		// const std::string export_surface = args["export"]["surface"];
 		// if (!export_surface.empty())
-		extract_vis_boundary_mesh();
+			extract_vis_boundary_mesh();
 		logger().info("Done!");
 
 		const int prev_b_size = local_boundary.size();
-		problem->setup_bc(*mesh, bases, local_boundary, boundary_nodes, local_neumann_boundary);
+		problem->setup_bc(*mesh, bases, pressure_bases, local_boundary, boundary_nodes, local_neumann_boundary, pressure_boundary_nodes);
```

Comment:
does this affects FEM

Reply:
No, this `pressure_boundary_nodes` is defined only to set up the pressure dirichlet bc for splitting. It's constructed in this function, same as what you do to construct the `boundary_nodes`.